src/app/caesar-portal/common/services/group.service.ts

```diff
@@ -2,21 +2,46 @@ import { Injectable } from '@angular/core';
 import { Http, Response, RequestOptions, Headers } from '@angular/http';
 import { environment } from '../../../../environments/environment';
 import { Group } from '../../common/models/group';
+import { User } from '../../common/models/user';
 import { Observable } from 'rxjs/Observable';
+import { Location } from '../../common/models/location';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/map';
 
 @Injectable()
 export class GroupService {
   private headers: Headers;
   private options: RequestOptions;
 
+  group: Group = new Group();
+  idCurrent = new BehaviorSubject<number>(null);
+
   constructor(private http: Http) {
     this.headers = new Headers({ 'Content-Type': 'application/json' });
     this.options = new RequestOptions({ headers: this.headers });
   }
 
-  get(id: number) {
-    return this.http.get(`${environment.serviceApi.groupsUrl}/${id}`);
+  get(id: any): Observable<Group> {
+    const search: URLSearchParams = new URLSearchParams();
+    search.set('id', id);
+
+    return this.http.get(`${environment.serviceApi.groupsUrl}/${id}`).
+      map(result => this.extractGroupsData(result.json()));
+  }
+
+  getLocation(url: string): Observable<Location> {
+    return this.getData(url);
+  }
+
+  getParametr(url: string): Observable<string[]> {
+    return this.getData(url);
+  }
+
+  getData(url: string) {
+    url = url.replace('http://caeser-api.com:8080', '');
```

Comment:
hmm... I don't like the idea with links received from BE at all, but maybe it's too late to change the API now, so let's keep it, maybe you can move it the host name to environment variables?

And you might need to add /api to the url so it'll be proxies as others

Reply:
Yes, this is not correct, but links to additional group's parameters are stored in other links that are stored in the group object. And now I cann't to solve the problem quickly. I can deal with this after the review, please, I do not want to let the guys down.